Sources/NIO/SocketChannel.swift

```diff
@@ -482,8 +482,13 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
             pipeline.fireErrorCaught0(error: err)
         }
 
-        executeAndComplete(promise) {
+        let p: EventLoopPromise<Void>?
+        do {
             try socket.close()
+            p = promise
+        } catch {
+            promise?.fail(error: error)
+            p = nil
```

Comment:
ok let me explain what I wanted to do ... Basically because we fail the promise here I wanted to ensure we pass `nil` to `becomeInactive0` so we not try to notify the promise again.

Reply:
Ah, ok. Please put a comment to that effect here.

Reply:
done